src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -1082,45 +1071,47 @@ public override SyntaxNode VisitInvocationExpression(InvocationExpressionSyntax
                     return rewrittenNode;
                 }
 
-                var expression = RewriteExtensionMethodInvocation(rewrittenNode, thisExpression, reducedExtensionMethod, s_typeNameFormatWithoutGenerics);
+                var expression = RewriteExtensionMethodInvocation(rewrittenNode, thisExpression, reducedExtensionMethod);
 
                 // Let's rebind this and verify the original method is being called properly
                 var binding = _semanticModel.GetSpeculativeSymbolInfo(originalNode.SpanStart, expression, SpeculativeBindingOption.BindAsExpression);
-
-                if (binding.Symbol != null)
-                {
-                    return expression;
-                }
-
-                // We'll probably need generic type arguments as well
-                return RewriteExtensionMethodInvocation(rewrittenNode, thisExpression, reducedExtensionMethod, s_typeNameFormatWithGenerics);
+                return binding.Symbol != null ? expression : originalNode;
```

Comment:
the line about `// We'll probably need generic type arguments as well` is bogus.  Extension methods can only be in top level static classes.  And static classes cannot be generic.  So there's no need to try to do anything with type-arguments in this code.